### PR TITLE
[BarClerk-553] - [Markup] Create Password reset page

### PR DESCRIPTION
**Pull request by impaulintech**
## Issue Link
- https://app.asana.com/0/1203234368773394/1203256158175553/f

## Definition of Done
- [ ] User can send a reset password link
- [ ] User can set a new password

## Pre-condition
- Input your email address
- Setup new password

## Expected Output
- When the user inputs the wrong credentials, it should return an error
- When the user inputs existing email in the database, they should receive an email confirmation link
- When they click the reset link, they should be redirected to the reset password page

## Screenshots/Recordings
image
image

### CODE REVIEW

File: client/components/atoms/Button/index.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ type Props = {
   isDisabled?: boolean
   onClick?: (value?: any) => void
   className?: string
```
Comment (abduljalilpalala) [inline client/components/atoms/Button/index.tsx:7]: Add replace `any` with its proper type.

File: client/components/templates/AdminAuthTemplate/index.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+const AdminAuthTemplate = ({ children, hasBorder = false }: any) => {
+  return (
```
Comment (abduljalilpalala) [inline client/components/templates/AdminAuthTemplate/index.tsx:4]: Add replace `any` with its proper type.

File: client/hooks/useRememberMe.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { Cookie } from "shared/types";
+import React, { useState } from "react";
+
+const useRememberMe = () => {
+  const [rememberMe, setRememberMe] = useState<string>("");
+
+  const rememberedEmail: Cookie = getCookie('email') || "";
+  const isRemembered: Cookie = getCookie('isRemembered') || false;
+
+  const onChangeRemember = (e: { target: any }) => {
```
Comment (abduljalilpalala) [inline client/hooks/useRememberMe.tsx:15]: Add replace `any` with its proper type.

File: client/hooks/useRememberMe.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    const { value, name } = e.target;
+    if (name === "password") return;
+
+    setRememberMe(value);
+    if (isRemembered) return setCookie('email', value);
+  }
+
+  const onClickRemember = (e: { target: any }) => {
+    const { checked } = e.target;
```
Comment (abduljalilpalala) [inline client/hooks/useRememberMe.tsx:24]: Add replace `any` with its proper type.

File: client/pages/forgot-password.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+} from "shared/validation";
+import Button from "components/atoms/Button";
+import NextHead from "components/atoms/NextHead";
+import { useAuthMethods } from "hooks/authMethods";
+import CustomForm from "components/molecules/CustomForm";
+import AdminAuthTemplate from "components/templates/AdminAuthTemplate";
+
+const ForgotPassword = () => {
+  const { state }: any = Router?.router || {};
+  const { user, verified } = state?.query || {};
```
Comment (abduljalilpalala) [inline client/pages/forgot-password.tsx:19]: Add replace `any` with its proper type.

File: client/pages/forgot-password.tsx [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  const setRequestResetLinkInitialValue = {
+    email: "",
+  };
+  const RequestResetLink = () => {
+    return (
+      <Formik
+        initialValues={setRequestResetLinkInitialValue}
+        validationSchema={ResetLinkSubmitFormSchema}
+        onSubmit={async ({ email }: { email: any; }) => {
+          await ResetLinkSubmit(email);
```
Comment (abduljalilpalala) [inline client/pages/forgot-password.tsx:37]: Add replace `any` with its proper type.

File: client/pages/forgot-password.tsx [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        }}
+      >
+        {({ isSubmitting }): any => {
+          return (
```
Comment (abduljalilpalala) [inline client/pages/forgot-password.tsx:41]: Add replace `any` with its proper type.

File: client/pages/forgot-password.tsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    password_confirmation: "",
+  };
+  const SetupNewPassword = (
+    <div className="flex flex-col">
+      <Formik
+        initialValues={setupNewPasswordInitialValue}
+        validationSchema={ForgotPasswordFormSchema}
+        onSubmit={handleSignInSubmit}
+      >
+        {({ isSubmitting }): any => {
```
Comment (abduljalilpalala) [inline client/pages/forgot-password.tsx:80]: Add replace `any` with its proper type.

**Review by abduljalilpalala — CHANGES_REQUESTED** [review]
Please check comments, also update the Pre-condition section of the PR. Add the steps on what the reviewer would do to use the feature.

**Comment by impaulintech** [comment]
All comments have been resolved, Thank you 🙇‍♂️

**Pull request merged**